lib/ja5_filter.h

```diff
@@ -0,0 +1,208 @@
+/**
+ *		Tempesta FW
+ *
+ * Copyright (C) 2024 Tempesta Technologies, Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License,
+ * or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program; if not, write to the Free Software Foundation, Inc., 59
+ * Temple Place - Suite 330, Boston, MA 02111-1307, USA.
+ */
+
+#include "log.h"
+#include "db/core/tdb.h"
+
+#include <linux/list.h>
+
+#define JA5_FILTER_TIME_SLOTS_CNT 8
+#define DB_RECS_TO_FREE_CNT 32
+
+typedef struct {
+	u32 counter;
+	u32 ts;
+} TimeSlot;
+
+typedef struct {
+	/** Keep  @list_node first for easy type casts */
+	struct list_head	list_node;
+	TimeSlot		conns[JA5_FILTER_TIME_SLOTS_CNT];
+	spinlock_t		conns_lock;
+	TimeSlot		recs[JA5_FILTER_TIME_SLOTS_CNT];
+	spinlock_t		recs_lock;
+} Rates;
+
+
+static struct {
+	TDB			*tdb;
+	struct list_head	lru_list;
+	spinlock_t		lru_list_lock;
+} storage;
+
+static bool
+get_alloc_ctx_eq_rec(TdbRec *, void *)
+{
+	return true;
```

Comment:
It seems that since we return always true here, we always return first entry from tdb. I mean that in `get_fingerprint_rates` we always catch first entry isn't it?

Reply:
Yes we always catch the first entry, and it's alright since we use fingerprints as keys, not hashes.

Reply:
But i think it's better to switch to hashed as keys